### PR TITLE
Allow preserving of original feature numbers (if not unique) in `combine_tobac_feats`

### PR DESCRIPTION
**Pull request by w-k-jones**
Addresses #296 

Add `renumber_features` option to combine_tobac_feats, which, if false, allows preserving of original feature numbers (if not unique)

Add `sort_features_by option` to allow user defined sort order of combined features

* [x] Have you followed our guidelines in CONTRIBUTING.md? 
* [x] Have you self-reviewed your code and corrected any misspellings? 
* [x] Have you written documentation that is easy to understand?
* [x] Have you written descriptive commit messages? 
* [x] Have you added NumPy docstrings for newly added functions? 
* [x] Have you formatted your code using black? 
* [x] If you have introduced a new functionality, have you added adequate unit tests?
* [x] Have all tests passed in your local clone? 
* [x] If you have introduced a new functionality, have you added an example notebook?
* [x] Have you kept your pull request small and limited so that it is easy to review? 
* [x] Have the newest changes from this branch been merged?

### CODE REVIEW

**Review by JuliaKukulies — APPROVED** [review]
Great @w-k-jones , I think you found a good compromise here with all the aspects that we discussed! I tested with some more data, all works fine and documentation is clear. Just a small suggestion for the error message to make it consistent with the docstrings.

File: tobac/utils/general.py [hunk 1]
```diff
@@ -539,15 +555,28 @@ def combine_tobac_feats(list_of_feats, preserve_old_feat_nums=None):
...
     # first, let's just combine these.
     combined_df = pd.concat(list_of_feats)
-    # Then, sort by time first, then by feature number
-    combined_df = combined_df.sort_values(["time", "feature"])
+
+    if not renumber_features and np.any(
+        np.bincount(combined_df["feature"] + np.nanmin(combined_df["feature"])) > 1
+    ):
+        raise ValueError(
+            "Non-unique feature values detected, please use renumber_features=True"
```
[the remark this answers is not in the document]
Comment (w-k-jones) [reply tobac/utils/general.py:563]: Thanks for the suggestion!

**Comment by w-k-jones** [comment]
@freemansw1 would you be happy for me to rename this function? I'd prefer to name it either `combine_features` or `combine_feature_dataframes` to make it a bit clearer

Also at some point we need to think again about where functions like this which are used as a postprocessing step, to separate them from internally used util functions

**Review by freemansw1 — APPROVED** [review]
Thanks for resolving this, @w-k-jones. This is a good and necessary change.

**Comment by w-k-jones** [comment]
@freemansw1 @JuliaKukulies I've now made the updates mentioned earlier. Let me know if you're happy for this to be merged

**Comment by JuliaKukulies** [comment]
That looks good @w-k-jones, ready to go!

**Pull request merged**